<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analyzing tRNA sequences and structures â€¢ tRNA</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analyzing tRNA sequences and structures">
<meta property="og:description" content="tRNA">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">tRNA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.7.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/tRNA.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/FelixErnst/tRNA/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="tRNA_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analyzing tRNA sequences and structures</h1>
                        <h4 class="author">Felix G.M. Ernst</h4>
            
            <h4 class="date">2020-07-18</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/FelixErnst/tRNA/blob/master/vignettes/tRNA.Rmd"><code>vignettes/tRNA.Rmd</code></a></small>
      <div class="hidden name"><code>tRNA.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Example of importing tRNAdb output as GRanges</p>
    </div>
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The <code>tRNA</code> package provides access to tRNA feature information for subsetting and visualization. Visualization functions are implemented to compare feature parameters of multiple tRNA sets and to correlate them to additional data.</p>
<p>As input the package expects a <code>GRanges</code> object with certain metadata columns. The following columns are required: <code>tRNA_length</code>, <code>tRNA_type</code>, <code>tRNA_anticodon</code>, <code>tRNA_seq</code>, <code>tRNA_str</code>, <code>tRNA_CCA.end</code>. The <code>tRNA_str</code> column must contain a valid dot bracket annotation. For more details please have a look at the vignette of the <code>Structstrings</code> package.</p>
</div>
<div id="loading-trna-information" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-trna-information" class="anchor"></a>Loading tRNA information</h1>
<p>To work with the <code>tRNA</code> package, tRNA information can be retrieved or loaded into a R session in a number of ways:</p>
<ol style="list-style-type: decimal">
<li>A <code>GRanges</code> object can be constructed manually containing the required colums mentioned above.</li>
<li>a tRNAscan result file can be loaded using the function <code>import.tRNAscanAsGRanges()</code> from the <code>tRNAscanImport</code> package</li>
<li>selected tRNA information can be retrieved using the function <code>import.tRNAdb()</code> from the <code>tRNAdbImport</code> package</li>
</ol>
<p>For the examples in this vignette a number of predefined <code>GRanges</code> objects are loaded.</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">tRNA</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/FelixErnst/Structstrings">Structstrings</a></span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"gr"</span>, package = <span class="st">"tRNA"</span>)
</pre></div>
</div>
<div id="trna-sequences-and-structures" class="section level1">
<h1 class="hasAnchor">
<a href="#trna-sequences-and-structures" class="anchor"></a>tRNA sequences and structures</h1>
<p>To retrieve the sequences for individual tRNA structure elements the functions <code>gettRNAstructureGRanges</code> or <code>gettRNAstructureSeqs</code> can be used. Several optional arguments can be used to modify the result (See <code><a href="../reference/gettRNAstructureGRanges.html">?gettRNAstructureSeqs</a></code>).</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="co"># just get the coordinates of the anticodonloop</span>
<span class="fu"><a href="../reference/gettRNAstructureGRanges.html">gettRNAstructureGRanges</a></span>(<span class="kw">gr</span>, structure = <span class="st">"anticodonLoop"</span>)
</pre></div>
<pre><code>## $anticodonLoop
## IRanges object with 299 ranges and 0 metadata columns:
##           start       end     width
##       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
##   TGG        31        37         7
##   TGC        32        38         7
##   CAA        31        37         7
##   AGA        31        37         7
##   TAA        31        37         7
##   ...       ...       ...       ...
##   CAT        32        38         7
##   GAA        31        37         7
##   TTA        31        37         7
##   TAC        32        38         7
##   CAT        32        38         7</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="fu"><a href="../reference/gettRNAstructureGRanges.html">gettRNAstructureSeqs</a></span>(<span class="kw">gr</span>, joinFeatures = <span class="fl">TRUE</span>, structure = <span class="st">"anticodonLoop"</span>)
</pre></div>
<pre><code>## $anticodonLoop
## RNAStringSet object of length 299:
##       width seq                                             names               
##   [1]     7 UUUGGGU                                         TGG
##   [2]     7 CUUGCAA                                         TGC
##   [3]     7 UUCAAGC                                         CAA
##   [4]     7 UUAGAAA                                         AGA
##   [5]     7 CUUAAGA                                         TAA
##   ...   ... ...
## [295]     7 CUCAUAA                                         CAT
## [296]     7 UUGAAGA                                         GAA
## [297]     7 UUUUAGU                                         TTA
## [298]     7 UUUACAC                                         TAC
## [299]     7 GUCAUGA                                         CAT</code></pre>
<p>In addition, the sequences can be returned already joined to get a fully blank padded set of sequences. The boundaries of the individual structures is returned as metadata of the <code>RNAStringSet</code> object.</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="kw">seqs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gettRNAstructureGRanges.html">gettRNAstructureSeqs</a></span>(<span class="kw">gr</span>[<span class="fl">1L</span><span class="op">:</span><span class="fl">10L</span>], joinCompletely = <span class="fl">TRUE</span>)
<span class="kw">seqs</span>
</pre></div>
<pre><code>## RNAStringSet object of length 10:
##      width seq
##  [1]    85 GGGCGUGUGGUC-UAGU-GGUAU-GAUUCUCGC...------GCCUGGGUUCAAUUCCCAGCUCGCCCC
##  [2]    85 GGGCACAUGGCGCAGUU-GGU-AGCGCGCUUCC...------GCAUCGGUUCGAUUCCGGUUGCGUCCA
##  [3]    85 GGUUGUUUGGCC-GAGC-GGUAA-GGCGCCUGA...AA-GAUGCAAGAGUUCGAAUCUCUUAGCAACCA
##  [4]    85 GGCAACUUGGCC-GAGU-GGUAA-GGCGAAAGA...U-GCCCGCGCAGGUUCGAGUCCUGCAGUUGUCG
##  [5]    85 GGAGGGUUGGCC-GAGU-GGUAA-GGCGGCAGA...UUGUCCGCGCGAGUUCGAACCUCGCAUCCUUCA
##  [6]    85 GCGGAUUUAGCUCAGUU-GGG-AGAGCGCCAGA...------GCCUGUGUUCGAUCCACAGAAUUCGCA
##  [7]    85 GGUCUCUUGGCC-CAGUUGGUAA-GGCACCGUG...------ACAGCGGUUCGAUCCCGCUAGAGACCA
##  [8]    85 GCGCAAGUGGUUUAGU--GGU-AAAAUCCAACG...-------CCCCGGUUCGAUUCCGGGCUUGCGCA
##  [9]    85 GGCAACUUGGCC-GAGU-GGUAA-GGCGAAAGA...U-GCCCGCGCAGGUUCGAGUCCUGCAGUUGUCG
## [10]    85 GCUUCUAUGGCC-AAGUUGGUAA-GGCGCCACA...------ACAUCGGUUCAAAUCCGAUUGGAAGCA</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="co"># getting the tRNA structure boundaries</span>
<span class="fu">metadata</span>(<span class="kw">seqs</span>)[[<span class="st">"tRNA_structures"</span>]]
</pre></div>
<pre><code>## IRanges object with 15 ranges and 0 metadata columns:
##                           start       end     width
##                       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
##   acceptorStem.prime5         1         7         7
##               Dprime5         8         9         2
##          DStem.prime5        10        13         4
##                 Dloop        14        23        10
##          DStem.prime3        24        27         4
##                   ...       ...       ...       ...
##          TStem.prime5        61        65         5
##                 Tloop        66        72         7
##          TStem.prime3        73        77         5
##   acceptorStem.prime3        78        84         7
##         discriminator        85        85         1</code></pre>
<p>Be aware, that <code>gettRNAstructureGRanges</code> and <code>gettRNAstructureSeqs</code> might not be working as expected, if the tRNA sequences in questions are armless or deviate drastically from the canonical tRNA model. The functions in the <code>tRNA</code> packages were thouroughly tested using human mitochondrial tRNA and other tRNAs missing certain features. However, for fringe cases results may differ. If you encounter such a case, please report it with an example.</p>
</div>
<div id="subsetting-trna-sequences" class="section level1">
<h1 class="hasAnchor">
<a href="#subsetting-trna-sequences" class="anchor"></a>Subsetting tRNA sequences</h1>
<p>Structure information of the tRNA can be queried for subsetting using several functions. For the following examples the functions <code>hasAccpeptorStem</code> and <code>hasDloop</code> are used.</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="kw">gr</span>[<span class="fu"><a href="../reference/hasTStem.html">hasAcceptorStem</a></span>(<span class="kw">gr</span>, unpaired = <span class="fl">TRUE</span>)]
<span class="co"># mismatches and bulged are subsets of unpaired</span>
<span class="kw">gr</span>[<span class="fu"><a href="../reference/hasTStem.html">hasAcceptorStem</a></span>(<span class="kw">gr</span>, mismatches = <span class="fl">TRUE</span>)]
<span class="kw">gr</span>[<span class="fu"><a href="../reference/hasTStem.html">hasAcceptorStem</a></span>(<span class="kw">gr</span>, bulged = <span class="fl">TRUE</span>)]
<span class="co"># combination of different structure parameters</span>
<span class="kw">gr</span>[<span class="fu"><a href="../reference/hasTStem.html">hasAcceptorStem</a></span>(<span class="kw">gr</span>, mismatches = <span class="fl">TRUE</span>) <span class="op">&amp;</span> 
     <span class="fu"><a href="../reference/hasTStem.html">hasDloop</a></span>(<span class="kw">gr</span>, length = <span class="fl">8L</span>)]
</pre></div>
<p>Please have a look at the man page <code>?hasAccpeptorStem</code> for all available subsetting functions.</p>
</div>
<div id="visualization" class="section level1">
<h1 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h1>
<p>To get an overview of tRNA features and compare different datasets, the function <code>gettRNAFeaturePlots</code> is used. It accepts a named <code>GRangesList</code> as input. Internally it will calculate a list of features values based on the functions mentioned above and the data contained in the mcols of the <code>GRanges</code> objects.</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="co"># load tRNA data for E. coli and H. sapiens</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"gr_eco"</span>, package = <span class="st">"tRNA"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"gr_human"</span>, package = <span class="st">"tRNA"</span>)

<span class="co"># get summary plots</span>
<span class="kw">grl</span> <span class="op">&lt;-</span> <span class="fu">GRangesList</span>(Sce = <span class="kw">gr</span>,
                   Hsa = <span class="kw">gr_human</span>,
                   Eco = <span class="kw">gr_eco</span>)
<span class="kw">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gettRNAFeaturePlots.html">gettRNAFeaturePlots</a></span>(<span class="kw">grl</span>)
</pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="kw">plots</span><span class="op">$</span><span class="kw">length</span>
</pre></div>
<div class="figure">
<img src="tRNA_files/figure-html/plot1-1.png" alt="tRNA length." width="700"><p class="caption">
tRNA length.
</p>
</div>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="kw">plots</span><span class="op">$</span><span class="kw">tRNAscan_score</span>
</pre></div>
<div class="figure">
<img src="tRNA_files/figure-html/plot2-1.png" alt="tRNAscan-SE scores." width="700"><p class="caption">
tRNAscan-SE scores.
</p>
</div>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="kw">plots</span><span class="op">$</span><span class="kw">gc</span>
</pre></div>
<div class="figure">
<img src="tRNA_files/figure-html/plot3-1.png" alt="tRNA GC content." width="700"><p class="caption">
tRNA GC content.
</p>
</div>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="kw">plots</span><span class="op">$</span><span class="kw">tRNAscan_intron</span>
</pre></div>
<div class="figure">
<img src="tRNA_files/figure-html/plot4-1.png" alt="tRNAs with introns." width="700"><p class="caption">
tRNAs with introns.
</p>
</div>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="kw">plots</span><span class="op">$</span><span class="kw">variableLoop_length</span>
</pre></div>
<div class="figure">
<img src="tRNA_files/figure-html/plot5-1.png" alt="Length of the variable loop." width="700"><p class="caption">
Length of the variable loop.
</p>
</div>
<p>To access the results without generating plots, use the function <code>gettRNASummary</code>.</p>
<p>To check whether features correlate with additional scores, optional arguments can be added to <code>gettRNAFeaturePlots</code> or used from the <code>score</code> column of the <code>GRanges</code> objects. For the first case a list of scores with the same dimensions as the <code>GRangesList</code> object has to be provided as the argument <code>scores</code>. For the latter case, just set the argument <code>plotScore = TRUE</code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="co"># score column will be used</span>
<span class="kw">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gettRNAFeaturePlots.html">gettRNAFeaturePlots</a></span>(<span class="kw">grl</span>, plotScores = <span class="fl">TRUE</span>)
</pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit">
<span class="kw">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gettRNAFeaturePlots.html">gettRNAFeaturePlots</a></span>(<span class="kw">grl</span>,
                             scores = <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">grl</span>[[<span class="fl">1L</span>]]),<span class="fl">0L</span>,<span class="fl">100L</span>),
                                           <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">grl</span>[[<span class="fl">2L</span>]]),<span class="fl">0L</span>,<span class="fl">100L</span>),
                                           <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">grl</span>[[<span class="fl">3L</span>]]),<span class="fl">0L</span>,<span class="fl">100L</span>)))
</pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="kw">plots</span><span class="op">$</span><span class="kw">length</span>
</pre></div>
<div class="figure">
<img src="tRNA_files/figure-html/plot6-1.png" alt="tRNA length and score correlation." width="700"><p class="caption">
tRNA length and score correlation.
</p>
</div>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="kw">plots</span><span class="op">$</span><span class="kw">variableLoop_length</span>
</pre></div>
<div class="figure">
<img src="tRNA_files/figure-html/plot7-1.png" alt="variable loop length and score correlation." width="700"><p class="caption">
variable loop length and score correlation.
</p>
</div>
<p>Since all plots returned by the functions mentioned above are <code>ggplot2</code> objects, they can be modified manually and changed to suit your needs.</p>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="kw">plots</span><span class="op">$</span><span class="kw">length</span><span class="op">$</span><span class="kw">layers</span> <span class="op">&lt;-</span> <span class="kw">plots</span><span class="op">$</span><span class="kw">length</span><span class="op">$</span><span class="kw">layers</span>[<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="op">-</span><span class="fl">1L</span>,<span class="op">-</span><span class="fl">2L</span>)]
<span class="kw">plots</span><span class="op">$</span><span class="kw">length</span> <span class="op">+</span> <span class="kw">ggplot2</span>::<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span>()
</pre></div>
<div class="figure">
<img src="tRNA_files/figure-html/plot8-1.png" alt="Customized plot switching out the point and violin plot into a boxplot." width="700"><p class="caption">
Customized plot switching out the point and violin plot into a boxplot.
</p>
</div>
<p>In addition, the data of the plots can be accessed directly.</p>
<div class="sourceCode" id="cb22"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">plots</span><span class="op">$</span><span class="kw">length</span><span class="op">$</span><span class="kw">data</span>)
</pre></div>
<pre><code>##    id value    score
## 1 Sce    71 12.33438
## 2 Sce    72 80.37610
## 3 Sce    80 59.73206
## 4 Sce    81 38.94058
## 5 Sce    82 96.24496
## 6 Sce    72 68.28609</code></pre>
</div>
<div id="options" class="section level1">
<h1 class="hasAnchor">
<a href="#options" class="anchor"></a>Options</h1>
<p>The colours of the plots can be customized directly on creation with the following options.</p>
<div class="sourceCode" id="cb24"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span>(<span class="st">"tRNA_colour_palette"</span>)
</pre></div>
<pre><code>## $tRNA_colour_palette
## [1] "Set1"</code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span>(<span class="st">"tRNA_colour_yes"</span>)
</pre></div>
<pre><code>## $tRNA_colour_yes
## [1] "green"</code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span>(<span class="st">"tRNA_colour_no"</span>)
</pre></div>
<pre><code>## $tRNA_colour_no
## [1] "red"</code></pre>
</div>
<div id="dot-bracket-annotation" class="section level1">
<h1 class="hasAnchor">
<a href="#dot-bracket-annotation" class="anchor"></a>Dot bracket annotation</h1>
<p>To retrieve detailed information on the base pairing the function <code><a href="../reference/gettRNABasePairing.html">gettRNABasePairing()</a></code> is used. Internally this will construct a <code>DotBracketStringSet</code> from the <code>tRNA_str</code> column, if this column does not already contain a <code>DotBracketStringSet</code>. It is then passed on to the <code><a href="https://rdrr.io/pkg/Structstrings/man/getBasePairing.html">Structstrings::getBasePairing</a></code> function.</p>
<p>A valid DotBracket annotation is expected to contain only pairs of <code>&lt;&gt;{}[]()</code> and the <code>.</code> character (Please note the orientation. For <code>&lt;&gt;</code> the orientation is variable, since the tRNAscan files use the <code>&gt;&lt;</code> annotation by default. However upon creation of a <code>DotBracketStringSet</code> this annotation will be converted).</p>
<div class="sourceCode" id="cb30"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu"><a href="../reference/gettRNABasePairing.html">gettRNABasePairing</a></span>(<span class="kw">gr</span>)[[<span class="fl">1L</span>]])
</pre></div>
<pre><code>## DotBracketDataFrame with 6 rows and 4 columns
##         pos   forward   reverse   character
##   &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt;
## 1         1         1        70           &lt;
## 2         2         2        69           &lt;
## 3         3         3        68           &lt;
## 4         4         4        67           &lt;
## 5         5         5        66           &lt;
## 6         6         0         0           .</code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Structstrings/man/getBasePairing.html">getBasePairing</a></span>(<span class="kw">gr</span>[<span class="fl">1L</span>]<span class="op">$</span><span class="kw">tRNA_str</span>)[[<span class="fl">1L</span>]])
</pre></div>
<pre><code>## DotBracketDataFrame with 6 rows and 4 columns
##         pos   forward   reverse   character
##   &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt;
## 1         1         1        70           &lt;
## 2         2         2        69           &lt;
## 3         3         3        68           &lt;
## 4         4         4        67           &lt;
## 5         5         5        66           &lt;
## 6         6         0         0           .</code></pre>
<p>The loop ids for the structure elements can be retrieved with the <code><a href="../reference/gettRNABasePairing.html">gettRNALoopIDs()</a></code> function, which relies on the <code><a href="https://rdrr.io/pkg/Structstrings/man/getBasePairing.html">Structstrings::getLoopIndices</a></code> function. (For more details, please have a look at the <code><a href="https://rdrr.io/pkg/Structstrings/man/getBasePairing.html">?getLoopIndices</a></code>)</p>
<div class="sourceCode" id="cb34"><pre class="downlit">
<span class="fu"><a href="../reference/gettRNABasePairing.html">gettRNALoopIDs</a></span>(<span class="kw">gr</span>)[[<span class="fl">1L</span>]]
</pre></div>
<pre><code>##  [1]  1  2  3  4  5  5  6  6  6  7  8  9  9  9  9  9  9  9  9  9  9  9  8  7  6
## [26] 10 11 12 13 14 14 14 14 14 14 14 14 14 13 12 11 10  6  6  6  6 15 16 17 18
## [51] 19 19 19 19 19 19 19 19 19 18 17 16 15  6  5  5  4  3  2  1  0</code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/Structstrings/man/getBasePairing.html">getLoopIndices</a></span>(<span class="kw">gr</span>[<span class="fl">1L</span>]<span class="op">$</span><span class="kw">tRNA_str</span>)
</pre></div>
<pre><code>## LoopIndexList of length 1
## [[""]] 1 2 3 4 5 5 6 6 6 7 8 9 9 9 9 9 ... 19 19 19 18 17 16 15 6 5 5 4 3 2 1 0</code></pre>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb38"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
</pre></div>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 18.04.4 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
## LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so
## 
## locale:
##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] tRNA_1.7.1           Structstrings_1.4.0  Biostrings_2.56.0   
##  [4] XVector_0.28.0       GenomicRanges_1.40.0 GenomeInfoDb_1.24.2 
##  [7] IRanges_2.22.2       S4Vectors_0.26.1     BiocGenerics_0.34.0 
## [10] BiocStyle_2.16.0    
## 
## loaded via a namespace (and not attached):
##  [1] assertive.properties_0.0-4   assertive.types_0.0-3       
##  [3] assertive.data.us_0.0-2      assertthat_0.2.1            
##  [5] rprojroot_1.3-2              digest_0.6.25               
##  [7] R6_2.4.1                     backports_1.1.8             
##  [9] evaluate_0.14                assertive.code_0.0-3        
## [11] ggplot2_3.3.2                highr_0.8                   
## [13] pillar_1.4.6                 assertive.strings_0.0-3     
## [15] zlibbioc_1.34.0              rlang_0.4.7                 
## [17] assertive_0.3-5.1            assertive.data_0.0-3        
## [19] rmarkdown_2.3.2              pkgdown_1.5.1.9000          
## [21] desc_1.2.0                   labeling_0.3                
## [23] stringr_1.4.0                RCurl_1.98-1.2              
## [25] munsell_0.5.0                compiler_4.0.2              
## [27] xfun_0.15                    pkgconfig_2.0.3             
## [29] htmltools_0.5.0              tibble_3.0.3                
## [31] GenomeInfoDbData_1.2.3       bookdown_0.20               
## [33] assertive.sets_0.0-3         codetools_0.2-16            
## [35] crayon_1.3.4                 withr_2.2.0                 
## [37] bitops_1.0-6                 grid_4.0.2                  
## [39] assertive.base_0.0-7         gtable_0.3.0                
## [41] lifecycle_0.2.0              magrittr_1.5                
## [43] assertive.models_0.0-2       scales_1.1.1                
## [45] stringi_1.4.6                farver_2.0.3                
## [47] fs_1.4.2                     assertive.matrices_0.0-2    
## [49] assertive.reflection_0.0-4.1 ellipsis_0.3.1              
## [51] assertive.datetimes_0.0-2    vctrs_0.3.2                 
## [53] RColorBrewer_1.1-2           tools_4.0.2                 
## [55] glue_1.4.1                   assertive.numbers_0.0-2     
## [57] yaml_2.2.1                   colorspace_1.4-1            
## [59] BiocManager_1.30.10          assertive.files_0.0-2       
## [61] assertive.data.uk_0.0-2      memoise_1.1.0               
## [63] knitr_1.29                   Modstrings_1.4.0</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Felix GM Ernst.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
