<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analyzing tRNA sequences and structures • tRNA</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analyzing tRNA sequences and structures">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">tRNA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.5.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/tRNA.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Analyzing tRNA sequences and structures</h1>
                        <h4 class="author">Felix G.M. Ernst</h4>
            
            <h4 class="date">2020-02-28</h4>
      
      
      <div class="hidden name"><code>tRNA.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      Example of importing tRNAdb output as GRanges
    </div>
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The <code>tRNA</code> package provides access to tRNA feature information for subsetting and visualization. Visualization functions are implemented to compare feature parameters of multiple tRNA sets and to correlate them to additional data.</p>
<p>As input the package expects a <code>GRanges</code> object with certain metadata columns. The following columns are required: <code>tRNA_length</code>, <code>tRNA_type</code>, <code>tRNA_anticodon</code>, <code>tRNA_seq</code>, <code>tRNA_str</code>, <code>tRNA_CCA.end</code>. The <code>tRNA_str</code> column must contain a valid dot bracket annotation. For more details please have a look at the vignette of the <code>Structstrings</code> package.</p>
</div>
<div id="loading-trna-information" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-trna-information" class="anchor"></a>Loading tRNA information</h1>
<p>To work with the <code>tRNA</code> package, tRNA information can be retrieved or loaded into a R session in a number of ways:</p>
<ol style="list-style-type: decimal">
<li>A <code>GRanges</code> object can be constructed manually containing the required colums mentioned above.</li>
<li>a tRNAscan result file can be loaded using the function <code>import.tRNAscanAsGRanges()</code> from the <code>tRNAscanImport</code> package</li>
<li>selected tRNA information can be retrieved using the function <code>import.tRNAdb()</code> from the <code>tRNAdbImport</code> package</li>
</ol>
<p>For the examples in this vignette a number of predefined <code>GRanges</code> objects are loaded.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(tRNA)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(Structstrings)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"gr"</span>, <span class="dt">package =</span> <span class="st">"tRNA"</span>)</a></code></pre></div>
</div>
<div id="trna-sequences-and-structures" class="section level1">
<h1 class="hasAnchor">
<a href="#trna-sequences-and-structures" class="anchor"></a>tRNA sequences and structures</h1>
<p>To retrieve the sequences for individual tRNA structure elements the functions <code>gettRNAstructureGRanges</code> or <code>gettRNAstructureSeqs</code> can be used. Several optional arguments can be used to modify the result (See <code><a href="../reference/gettRNAstructureSeqs.html">?gettRNAstructureSeqs</a></code>).</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># just get the coordinates of the anticodonloop</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="../reference/gettRNAstructureSeqs.html">gettRNAstructureGRanges</a></span>(gr, <span class="dt">structure =</span> <span class="st">"anticodonLoop"</span>)</a></code></pre></div>
<pre><code>## $anticodonLoop
## IRanges object with 299 ranges and 0 metadata columns:
##           start       end     width
##       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
##   TGG        31        37         7
##   TGC        32        38         7
##   CAA        31        37         7
##   AGA        31        37         7
##   TAA        31        37         7
##   ...       ...       ...       ...
##   CAT        32        38         7
##   GAA        31        37         7
##   TTA        31        37         7
##   TAC        32        38         7
##   CAT        32        38         7</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="../reference/gettRNAstructureSeqs.html">gettRNAstructureSeqs</a></span>(gr, <span class="dt">joinFeatures =</span> <span class="ot">TRUE</span>, <span class="dt">structure =</span> <span class="st">"anticodonLoop"</span>)</a></code></pre></div>
<pre><code>## $anticodonLoop
## RNAStringSet object of length 299:
##       width seq                                             names               
##   [1]     7 UUUGGGU                                         TGG
##   [2]     7 CUUGCAA                                         TGC
##   [3]     7 UUCAAGC                                         CAA
##   [4]     7 UUAGAAA                                         AGA
##   [5]     7 CUUAAGA                                         TAA
##   ...   ... ...
## [295]     7 CUCAUAA                                         CAT
## [296]     7 UUGAAGA                                         GAA
## [297]     7 UUUUAGU                                         TTA
## [298]     7 UUUACAC                                         TAC
## [299]     7 GUCAUGA                                         CAT</code></pre>
<p>In addition, the sequences can be returned already joined to get a fully blank padded set of sequences. The boundaries of the individual structures is returned as metadata of the <code>RNAStringSet</code> object.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">seqs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gettRNAstructureSeqs.html">gettRNAstructureSeqs</a></span>(gr[1L<span class="op">:</span>10L], <span class="dt">joinCompletely =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">seqs</a></code></pre></div>
<pre><code>## RNAStringSet object of length 10:
##      width seq
##  [1]    85 GGGCGUGUGGUC-UAGU-GGUAU-GAUUCUCGCU…-------GCCUGGGUUCAAUUCCCAGCUCGCCCC
##  [2]    85 GGGCACAUGGCGCAGUU-GGU-AGCGCGCUUCCC…-------GCAUCGGUUCGAUUCCGGUUGCGUCCA
##  [3]    85 GGUUGUUUGGCC-GAGC-GGUAA-GGCGCCUGAU…UAA-GAUGCAAGAGUUCGAAUCUCUUAGCAACCA
##  [4]    85 GGCAACUUGGCC-GAGU-GGUAA-GGCGAAAGAU…UU-GCCCGCGCAGGUUCGAGUCCUGCAGUUGUCG
##  [5]    85 GGAGGGUUGGCC-GAGU-GGUAA-GGCGGCAGAC…GUUGUCCGCGCGAGUUCGAACCUCGCAUCCUUCA
##  [6]    85 GCGGAUUUAGCUCAGUU-GGG-AGAGCGCCAGAC…-------GCCUGUGUUCGAUCCACAGAAUUCGCA
##  [7]    85 GGUCUCUUGGCC-CAGUUGGUAA-GGCACCGUGC…-------ACAGCGGUUCGAUCCCGCUAGAGACCA
##  [8]    85 GCGCAAGUGGUUUAGU--GGU-AAAAUCCAACGU…--------CCCCGGUUCGAUUCCGGGCUUGCGCA
##  [9]    85 GGCAACUUGGCC-GAGU-GGUAA-GGCGAAAGAU…UU-GCCCGCGCAGGUUCGAGUCCUGCAGUUGUCG
## [10]    85 GCUUCUAUGGCC-AAGUUGGUAA-GGCGCCACAC…-------ACAUCGGUUCAAAUCCGAUUGGAAGCA</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co"># getting the tRNA structure boundaries</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw">metadata</span>(seqs)[[<span class="st">"tRNA_structures"</span>]]</a></code></pre></div>
<pre><code>## IRanges object with 15 ranges and 0 metadata columns:
##                           start       end     width
##                       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
##   acceptorStem.prime5         1         7         7
##               Dprime5         8         9         2
##          DStem.prime5        10        13         4
##                 Dloop        14        23        10
##          DStem.prime3        24        27         4
##                   ...       ...       ...       ...
##          TStem.prime5        61        65         5
##                 Tloop        66        72         7
##          TStem.prime3        73        77         5
##   acceptorStem.prime3        78        84         7
##         discriminator        85        85         1</code></pre>
<p>Be aware, that <code>gettRNAstructureGRanges</code> and <code>gettRNAstructureSeqs</code> might not be working as expected, if the tRNA sequences in questions are armless or deviate drastically from the canonical tRNA model. The functions in the <code>tRNA</code> packages were thouroughly tested using human mitochondrial tRNA and other tRNAs missing certain features. However, for fringe cases results may differ. If you encounter such a case, please report it with an example.</p>
</div>
<div id="subsetting-trna-sequences" class="section level1">
<h1 class="hasAnchor">
<a href="#subsetting-trna-sequences" class="anchor"></a>Subsetting tRNA sequences</h1>
<p>Structure information of the tRNA can be queried for subsetting using several functions. For the following examples the functions <code>hasAccpeptorStem</code> and <code>hasDloop</code> are used.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">gr[<span class="kw"><a href="../reference/tRNA-subset.html">hasAcceptorStem</a></span>(gr, <span class="dt">unpaired =</span> <span class="ot">TRUE</span>)]</a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="co"># mismatches and bulged are subsets of unpaired</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3">gr[<span class="kw"><a href="../reference/tRNA-subset.html">hasAcceptorStem</a></span>(gr, <span class="dt">mismatches =</span> <span class="ot">TRUE</span>)]</a>
<a class="sourceLine" id="cb10-4" data-line-number="4">gr[<span class="kw"><a href="../reference/tRNA-subset.html">hasAcceptorStem</a></span>(gr, <span class="dt">bulged =</span> <span class="ot">TRUE</span>)]</a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="co"># combination of different structure parameters</span></a>
<a class="sourceLine" id="cb10-6" data-line-number="6">gr[<span class="kw"><a href="../reference/tRNA-subset.html">hasAcceptorStem</a></span>(gr, <span class="dt">mismatches =</span> <span class="ot">TRUE</span>) <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="st">     </span><span class="kw"><a href="../reference/tRNA-subset.html">hasDloop</a></span>(gr, <span class="dt">length =</span> 8L)]</a></code></pre></div>
<p>Please have a look at the man page <code>?hasAccpeptorStem</code> for all available subsetting functions.</p>
</div>
<div id="visualization" class="section level1">
<h1 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h1>
<p>To get an overview of tRNA features and compare different datasets, the function <code>gettRNAFeaturePlots</code> is used. It accepts a named <code>GRangesList</code> as input. Internally it will calculate a list of features values based on the functions mentioned above and the data contained in the mcols of the <code>GRanges</code> objects.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="co"># load tRNA data for E. coli and H. sapiens</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"gr_eco"</span>, <span class="dt">package =</span> <span class="st">"tRNA"</span>)</a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"gr_human"</span>, <span class="dt">package =</span> <span class="st">"tRNA"</span>)</a>
<a class="sourceLine" id="cb11-4" data-line-number="4"></a>
<a class="sourceLine" id="cb11-5" data-line-number="5"><span class="co"># get summary plots</span></a>
<a class="sourceLine" id="cb11-6" data-line-number="6">grl &lt;-<span class="st"> </span><span class="kw">GRangesList</span>(<span class="dt">Sce =</span> gr,</a>
<a class="sourceLine" id="cb11-7" data-line-number="7">                   <span class="dt">Hsa =</span> gr_human,</a>
<a class="sourceLine" id="cb11-8" data-line-number="8">                   <span class="dt">Eco =</span> gr_eco)</a>
<a class="sourceLine" id="cb11-9" data-line-number="9">plots &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gettRNAFeaturePlots.html">gettRNAFeaturePlots</a></span>(grl)</a></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">plots<span class="op">$</span>length</a></code></pre></div>
<div class="figure">
<img src="tRNA_files/figure-html/plot1-1.png" alt="tRNA length." width="700"><p class="caption">
tRNA length.
</p>
</div>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">plots<span class="op">$</span>tRNAscan_score</a></code></pre></div>
<div class="figure">
<img src="tRNA_files/figure-html/plot2-1.png" alt="tRNAscan-SE scores." width="700"><p class="caption">
tRNAscan-SE scores.
</p>
</div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">plots<span class="op">$</span>gc</a></code></pre></div>
<div class="figure">
<img src="tRNA_files/figure-html/plot3-1.png" alt="tRNA GC content." width="700"><p class="caption">
tRNA GC content.
</p>
</div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">plots<span class="op">$</span>tRNAscan_intron</a></code></pre></div>
<div class="figure">
<img src="tRNA_files/figure-html/plot4-1.png" alt="tRNAs with introns." width="700"><p class="caption">
tRNAs with introns.
</p>
</div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">plots<span class="op">$</span>variableLoop_length</a></code></pre></div>
<div class="figure">
<img src="tRNA_files/figure-html/plot5-1.png" alt="Length of the variable loop." width="700"><p class="caption">
Length of the variable loop.
</p>
</div>
<p>To access the results without generating plots, use the function <code>gettRNASummary</code>.</p>
<p>To check whether features correlate with additional scores, optional arguments can be added to <code>gettRNAFeaturePlots</code> or used from the <code>score</code> column of the <code>GRanges</code> objects. For the first case a list of scores with the same dimensions as the <code>GRangesList</code> object has to be provided as the argument <code>scores</code>. For the latter case, just set the argument <code>plotScore = TRUE</code>.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="co"># score column will be used</span></a>
<a class="sourceLine" id="cb17-2" data-line-number="2">plots &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gettRNAFeaturePlots.html">gettRNAFeaturePlots</a></span>(grl, <span class="dt">plotScores =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">plots &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gettRNAFeaturePlots.html">gettRNAFeaturePlots</a></span>(grl,</a>
<a class="sourceLine" id="cb18-2" data-line-number="2">                             <span class="dt">scores =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(grl[[1L]]),0L,100L),</a>
<a class="sourceLine" id="cb18-3" data-line-number="3">                                           <span class="kw"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(grl[[2L]]),0L,100L),</a>
<a class="sourceLine" id="cb18-4" data-line-number="4">                                           <span class="kw"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(grl[[3L]]),0L,100L)))</a></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">plots<span class="op">$</span>length</a></code></pre></div>
<div class="figure">
<img src="tRNA_files/figure-html/plot6-1.png" alt="tRNA length and score correlation." width="700"><p class="caption">
tRNA length and score correlation.
</p>
</div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">plots<span class="op">$</span>variableLoop_length</a></code></pre></div>
<div class="figure">
<img src="tRNA_files/figure-html/plot7-1.png" alt="variable loop length and score correlation." width="700"><p class="caption">
variable loop length and score correlation.
</p>
</div>
<p>Since all plots returned by the functions mentioned above are <code>ggplot2</code> objects, they can be modified manually and changed to suit your needs.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">plots<span class="op">$</span>length<span class="op">$</span>layers &lt;-<span class="st"> </span>plots<span class="op">$</span>length<span class="op">$</span>layers[<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="op">-</span>1L,<span class="op">-</span>2L)]</a>
<a class="sourceLine" id="cb21-2" data-line-number="2">plots<span class="op">$</span>length <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span>()</a></code></pre></div>
<div class="figure">
<img src="tRNA_files/figure-html/plot8-1.png" alt="Customized plot switching out the point and violin plot into a boxplot." width="700"><p class="caption">
Customized plot switching out the point and violin plot into a boxplot.
</p>
</div>
<p>In addition, the data of the plots can be accessed directly.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(plots<span class="op">$</span>length<span class="op">$</span>data)</a></code></pre></div>
<pre><code>##    id value    score
## 1 Sce    71 43.19781
## 2 Sce    72 52.28045
## 3 Sce    80 31.30666
## 4 Sce    81 81.24785
## 5 Sce    82 28.23450
## 6 Sce    72 81.77661</code></pre>
</div>
<div id="options" class="section level1">
<h1 class="hasAnchor">
<a href="#options" class="anchor"></a>Options</h1>
<p>The colours of the plots can be customized directly on creation with the following options.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/options.html">options</a></span>(<span class="st">"tRNA_colour_palette"</span>)</a></code></pre></div>
<pre><code>## $tRNA_colour_palette
## [1] "Set1"</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/options.html">options</a></span>(<span class="st">"tRNA_colour_yes"</span>)</a></code></pre></div>
<pre><code>## $tRNA_colour_yes
## [1] "green"</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/options.html">options</a></span>(<span class="st">"tRNA_colour_no"</span>)</a></code></pre></div>
<pre><code>## $tRNA_colour_no
## [1] "red"</code></pre>
</div>
<div id="dot-bracket-annotation" class="section level1">
<h1 class="hasAnchor">
<a href="#dot-bracket-annotation" class="anchor"></a>Dot bracket annotation</h1>
<p>To retrieve detailed information on the base pairing the function <code><a href="../reference/gettRNABasePairing.html">gettRNABasePairing()</a></code> is used. Internally this will construct a <code>DotBracketStringSet</code> from the <code>tRNA_str</code> column, if this column does not already contain a <code>DotBracketStringSet</code>. It is then passed on to the <code><a href="https://rdrr.io/pkg/Structstrings/man/getBasePairing.html">Structstrings::getBasePairing</a></code> function.</p>
<p>A valid DotBracket annotation is expected to contain only pairs of <code>&lt;&gt;{}[]()</code> and the <code>.</code> character (Please note the orientation. For <code>&lt;&gt;</code> the orientation is variable, since the tRNAscan files use the <code>&gt;&lt;</code> annotation by default. However upon creation of a <code>DotBracketStringSet</code> this annotation will be converted).</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw"><a href="../reference/gettRNABasePairing.html">gettRNABasePairing</a></span>(gr)[[1L]])</a></code></pre></div>
<pre><code>## DotBracketDataFrame with 6 rows and 4 columns
##         pos   forward   reverse   character
##   &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt;
## 1         1         1        70           &lt;
## 2         2         2        69           &lt;
## 3         3         3        68           &lt;
## 4         4         4        67           &lt;
## 5         5         5        66           &lt;
## 6         6         0         0           .</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Structstrings/man/getBasePairing.html">getBasePairing</a></span>(gr[1L]<span class="op">$</span>tRNA_str)[[1L]])</a></code></pre></div>
<pre><code>## DotBracketDataFrame with 6 rows and 4 columns
##         pos   forward   reverse   character
##   &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt;
## 1         1         1        70           &lt;
## 2         2         2        69           &lt;
## 3         3         3        68           &lt;
## 4         4         4        67           &lt;
## 5         5         5        66           &lt;
## 6         6         0         0           .</code></pre>
<p>The loop ids for the structure elements can be retrieved with the <code><a href="../reference/gettRNABasePairing.html">gettRNALoopIDs()</a></code> function, which relies on the <code><a href="https://rdrr.io/pkg/Structstrings/man/getBasePairing.html">Structstrings::getLoopIndices</a></code> function. (For more details, please have a look at the <code><a href="https://rdrr.io/pkg/Structstrings/man/getBasePairing.html">?getLoopIndices</a></code>)</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1"><span class="kw"><a href="../reference/gettRNABasePairing.html">gettRNALoopIDs</a></span>(gr)[[1L]]</a></code></pre></div>
<pre><code>##  [1]  1  2  3  4  5  5  6  6  6  7  8  9  9  9  9  9  9  9  9  9  9  9  8  7  6
## [26] 10 11 12 13 14 14 14 14 14 14 14 14 14 13 12 11 10  6  6  6  6 15 16 17 18
## [51] 19 19 19 19 19 19 19 19 19 18 17 16 15  6  5  5  4  3  2  1  0</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/pkg/Structstrings/man/getBasePairing.html">getLoopIndices</a></span>(gr[1L]<span class="op">$</span>tRNA_str)</a></code></pre></div>
<pre><code>## LoopIndexList of length 1
## [[""]] 1 2 3 4 5 5 6 6 6 7 8 9 9 9 9 9 ... 19 19 19 18 17 16 15 6 5 5 4 3 2 1 0</code></pre>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</a></code></pre></div>
<pre><code>## R Under development (unstable) (2020-02-28 r77869)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 16.04.6 LTS
## 
## Matrix products: default
## BLAS:   /home/travis/R-bin/lib/R/lib/libRblas.so
## LAPACK: /home/travis/R-bin/lib/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] tRNA_1.5.2           Structstrings_1.3.3  Biostrings_2.55.4   
##  [4] XVector_0.27.0       GenomicRanges_1.39.2 GenomeInfoDb_1.23.13
##  [7] IRanges_2.21.3       S4Vectors_0.25.12    BiocGenerics_0.33.0 
## [10] BiocStyle_2.15.6    
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.3                 assertive.properties_0.0-4
##  [3] assertive.types_0.0-3      assertive.data.us_0.0-2   
##  [5] assertthat_0.2.1           rprojroot_1.3-2           
##  [7] digest_0.6.25              R6_2.4.1                  
##  [9] backports_1.1.5            evaluate_0.14             
## [11] assertive.code_0.0-3       highr_0.8                 
## [13] ggplot2_3.2.1              pillar_1.4.3              
## [15] assertive.strings_0.0-3    zlibbioc_1.33.1           
## [17] rlang_0.4.4                lazyeval_0.2.2            
## [19] assertive_0.3-5            assertive.data_0.0-3      
## [21] rmarkdown_2.1              pkgdown_1.4.1             
## [23] labeling_0.3               desc_1.2.0                
## [25] stringr_1.4.0              RCurl_1.98-1.1            
## [27] munsell_0.5.0              compiler_4.0.0            
## [29] xfun_0.12                  pkgconfig_2.0.3           
## [31] htmltools_0.4.0            tibble_2.1.3              
## [33] GenomeInfoDbData_1.2.2     bookdown_0.17             
## [35] assertive.sets_0.0-3       codetools_0.2-16          
## [37] crayon_1.3.4               withr_2.1.2               
## [39] MASS_7.3-51.5              bitops_1.0-6              
## [41] grid_4.0.0                 assertive.base_0.0-7      
## [43] gtable_0.3.0               lifecycle_0.1.0           
## [45] magrittr_1.5               assertive.models_0.0-2    
## [47] scales_1.1.0               stringi_1.4.6             
## [49] farver_2.0.3               fs_1.3.1                  
## [51] assertive.matrices_0.0-2   assertive.reflection_0.0-4
## [53] assertive.datetimes_0.0-2  RColorBrewer_1.1-2        
## [55] tools_4.0.0                assertive.numbers_0.0-2   
## [57] yaml_2.2.1                 colorspace_1.4-1          
## [59] BiocManager_1.30.10        assertive.files_0.0-2     
## [61] assertive.data.uk_0.0-2    memoise_1.1.0             
## [63] knitr_1.28                 Modstrings_1.3.6</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#loading-trna-information">Loading tRNA information</a></li>
      <li><a href="#trna-sequences-and-structures">tRNA sequences and structures</a></li>
      <li><a href="#subsetting-trna-sequences">Subsetting tRNA sequences</a></li>
      <li><a href="#visualization">Visualization</a></li>
      <li><a href="#options">Options</a></li>
      <li><a href="#dot-bracket-annotation">Dot bracket annotation</a></li>
      <li><a href="#session-info">Session info</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Felix GM Ernst.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
